reader.go

```diff
@@ -783,6 +787,55 @@ func (r *Reader) commitLoop(conn *Conn) func(stop <-chan struct{}) {
 	}
 }
 
+// partitionWatcher queries kafka and watches for partition changes, triggering a rebalance if changes are found.
+func (r *Reader) partitionWatcher(conn *Conn) func(stop <-chan struct{}) {
+	return func(stop <-chan struct{}) {
+		ticker := time.NewTicker(defaultPartitionWatchTime)
+		defer ticker.Stop()
+		ops, err := conn.ReadPartitions(r.config.Topic)
+		if err != nil {
+			r.withErrorLogger(func(l *log.Logger) {
+				l.Printf("Problem getting partitions during startup, %v\n, Returning and setting up handshake", err)
+			})
+			return
+		}
+		// It's possible that the list of partitions returned are not in any order
+		// so put them into a map to compare.
+		oParts := make(map[int]struct{})
```

Comment:
I don't think you need a map here since we believe that Kafka won't return a partition range with holes.  You can use a slice instead to avoid a heap allocation:
```suggestion
		oParts := make([]int, len(ops))
		for i := range oParts {
			oParts[i] = i
		}
```

Reply:
I can do that, I was using the map to ensure that it's sorted as well, because I didn't want to do the array each time.

Reply:
If I'm only checking the length tho, the array should be fine.